packages/components/src/Bottega.purs

```diff
@@ -30,12 +32,19 @@ getOrder { userId, authToken } orderNumber = do
     authorization = oauthToken authToken
     authUser = unsafeToForeign userId
 
-readOrder :: { number :: OrderNumber, user :: UUID, status :: { state :: Foreign, time :: String } } -> Aff Order
+readOrder :: { number :: OrderNumber, user :: UUID, status :: { state :: Foreign, time :: String, failReason :: Nullable Foreign } } -> Aff Order
```

Comment:
I think `failReason` can just be `Nullable String`

Reply:
But then how do I parse an object of type `OrderStatusFailReason` out of it?

Reply:
Hmm, `state` seems to be `Foreign` as well 🤔 Should be `String` too.

Reply:
Basically you are already doing it here
```
    orderStatusStateString :: String <- readImpl foreignOrderStatusState
    case orderStatusStateString of
      "created"   -> pure OrderCreated
      "started"   -> pure OrderStarted
      "completed" -> pure OrderCompleted
      "failed"    -> pure OrderFailed
      "canceled"  -> pure OrderCanceled
      _           -> pure UnknownState
```

Reply:
But wait, the signature of the `readImpl` function is `readImpl :: Foreign -> F a`: 
https://pursuit.purescript.org/packages/purescript-simple-json/7.0.0/docs/Simple.JSON#v:readImpl

So isn't `Nullable Foreign` correct?